pkg/k8s/mounterBuilder.go

```diff
@@ -0,0 +1,129 @@
+package k8s
+
+import (
+	"github.com/feedhenry/mcp-standalone/pkg/mobile"
+	"github.com/pkg/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+type MounterBuilder struct {
+	k8s       kubernetes.Interface
+	namespace string
+}
+
+func NewMounterBuilder(namespace string) mobile.MounterBuilder {
+	return &MounterBuilder{namespace: namespace}
+}
+
+func (mb *MounterBuilder) WithK8s(k8s kubernetes.Interface) mobile.MounterBuilder {
```

Comment:
one thing I am not sure on here. As we are passing in  a kubernetes client, we want to avoid accidentally reusing that client. As it is stored in the struct if something else was using this, it could have undesired behaviour where the client is changed by one request and then another calls build and gets the wrong client. To avoid this, I think it is better to always create a new MountBuilder and return that rather than returning the current instance. 
``` return &MountBuilder{k8s:k8s,namespace:namespace} that way you have a new pointer for each action and different threads cannot interfere.